pt_miniscreen/app.py

```diff
@@ -88,59 +98,63 @@ def setup_events(self) -> None:
         )
         subscribe(AppEvent.STOP_SCREENSAVER, lambda: self.pop_tile_group())
 
-        def set_is_user_controlled(user_has_control) -> None:
-            if self.user_has_control and not user_has_control:
-                self.user_gave_back_control_event.set()
+    def set_is_user_controlled(self, user_has_control) -> None:
+        if self.user_has_control and not user_has_control:
+            self.user_gave_back_control_event.set()
+
+        logger.info(
+            f"User has {'taken' if user_has_control else 'given back'} control of the miniscreen"
+        )
+
+    def on_button_released(self):
+        if not self.state_manager.buttons_should_be_handled():
+            logger.info("State manager says that buttons should not be handled")
+            return
+
+        self.state_manager.wake()
+
+    def on_up_button_released(self):
+        logger.debug(f"Handling up button for tile group {self.current_tile_group}")
+        self.on_button_released()
 
-            logger.info(
-                f"User has {'taken' if user_has_control else 'given back'} control of the miniscreen"
-            )
+        current_tile = self.current_tile_group.current_menu_tile
+        if isinstance(current_tile, PaginatedTile):
+            current_tile.scroll_to_previous()
 
-        self.miniscreen.when_user_controlled = lambda: set_is_user_controlled(True)
-        self.miniscreen.when_system_controlled = lambda: set_is_user_controlled(False)
+    def on_down_button_released(self):
+        logger.debug(f"Handling down button for tile group {self.current_tile_group}")
+        self.on_button_released()
 
-        class MiniscreenButton(Enum):
-            UP = auto()
-            DOWN = auto()
-            SELECT = auto()
-            CANCEL = auto()
+        current_tile = self.current_tile_group.current_menu_tile
+        if isinstance(current_tile, PaginatedTile):
+            current_tile.scroll_to_next()
 
-        def handle_button_press(button: MiniscreenButton):
-            logger.debug(
-                f"Handling button {button} for tile group {self.current_tile_group}"
-            )
+    def on_select_button_released(self):
+        logger.debug(f"Handling select button for tile group {self.current_tile_group}")
+        self.on_button_released()
 
-            if not self.state_manager.buttons_should_be_handled():
-                logger.info("State manager says that buttons should not be handled")
+        current_tile = self.current_tile_group.current_menu_tile
+        if isinstance(current_tile, MenuTile):
+            handled = current_tile.go_to_child_menu()
+            if handled:
                 return
 
-            self.state_manager.wake()
+        # if menu doesn't have a child menu it might be an action page
+        current_tile.start_action()
```

Comment:
yeah, i think this child or action this should be implemented in the menu `on_select`, not here in the app